libs/wingc/src/jsify.rs

```diff
@@ -591,6 +591,59 @@ impl<'a> JSifier<'a> {
 			}
 			StmtKind::Break => CodeMaker::one_line("break;"),
 			StmtKind::Continue => CodeMaker::one_line("continue;"),
+			StmtKind::IfLet {
+				value,
+				statements,
+				var_name,
+				else_statements,
+			} => {
+				let mut code = CodeMaker::default();
+				// To enable shadowing variables in if let statements, the following does some scope trickery
```

Comment:
I took some time to think about how to handle shadowing here, I think this is the easiest way but let me know whatcha think.

Reply:
Makes sense to me 👍 . Another way could be
```js
const x = "hello"
if (x != undefined) {
    (($IF_LET_VARIABLE) => {
        const x = $IF_LET_VARIABLE;
        console.log(x);
    })(x);
}
```
(I'm not sure if there's any functional difference though, so we can probably keep the existing way)

Reply:
@Chriscbr I love that, wish I would have thought about it 😂  let me check if it works (which Im sure it does) Ill update PR if so. I dont believe there is a functional difference but as far as conciseness is concerned ❤️

Reply:
Ah actually this causes some issues with returning out of an `if let` statement. Ill keep the current impl for now